Source/Generators/LMCCavitySignalGenerator.cc

```diff
@@ -579,29 +574,29 @@ namespace locust
     		{
     			for (int n=0; n<fNModes; n++)
     			{
-    				if (ModeSelect(l, m, n, fE_Gun))
+    				if (ModeSelect(l, m, n, fInterface->fE_Gun))
     				{
-    			    	std::vector<double> tE_normalized;
-						tE_normalized = fInterface->fField->GetNormalizedModeField(l,m,n,tKassParticleXP);
+    					std::vector<double> tE_normalized;
+    					tE_normalized = fInterface->fField->GetNormalizedModeField(l,m,n,tKassParticleXP);
     					double cavityFIRSample = aFieldCalculator.GetCavityFIRSample(tKassParticleXP, fBypassTF);
-    			    	dopplerFrequency = fInterface->fField->GetDopplerFrequency(l, m, n, tKassParticleXP);
-						fAvgDotProductFactor = 1. / ( tThisEventNSamples + 1 ) * ( fAvgDotProductFactor * tThisEventNSamples + fInterface->fField->GetDotProductFactor(tKassParticleXP, tE_normalized, fIntermediateFile) );  // unit velocity \dot unit theta
+    					dopplerFrequency = fInterface->fField->GetDopplerFrequency(l, m, n, tKassParticleXP);
+    					fAvgDotProductFactor = 1. / ( tThisEventNSamples + 1 ) * ( fAvgDotProductFactor * tThisEventNSamples + fInterface->fField->GetDotProductFactor(tKassParticleXP, tE_normalized, fIntermediateFile) );  // unit velocity \dot unit theta
     					double modeAmplitude = tE_normalized.back();
 
-    					if (!fE_Gun)
+    					if (!fInterface->fE_Gun)
     					{
-    						unitConversion = 1.;  // mks units in Collin amplitudes.
-    						excitationAmplitude = fAvgDotProductFactor * modeAmplitude * cavityFIRSample;
+    						// sqrt(4PIeps0) for Kass current si->cgs, sqrt(4PIeps0) for A_lambda coefficient cgs->si
+    						unitConversion = 1. / LMCConst::FourPiEps(); // see comment
+    						excitationAmplitude = fAvgDotProductFactor * modeAmplitude * cavityFIRSample * fInterface->fField->Z_TE(l,m,n,tKassParticleXP[7]) * 2. * LMCConst::Pi() / LMCConst::C() / 1.e2;
     						std::vector<double> tProbeLocation = {fInterface->fField->GetDimR()*fPowerCombiner->GetCavityProbeRFrac(), 0., fPowerCombiner->GetCavityProbeZ()};

```

Comment:
I know this push has some changes between cgs and SI units and I always get confused in those differences, but at least at first glance there apparently seems to be different units here vs the fE_gun "else" block definition of the `excitationAmplitude` variable. Everything is identical besides the Z_TE (in ohms?) and the ScaleEPoyntingVector (in sqrt of power flux?) here. I don't think this really matters at this stage due to the arbitrary wave impedance value anyway, but it's worth keeping in mind.

Reply:
There might be some more ways to check this, such as not calculating a Poynting vector and instead driving the probe with the normalized E-field, and then multiplying the Katydid output by the cross-secional area of the waveguide.  Since we moved to a complex FIR there is still some normalization that needs a closer look.